packages/devtools_app/lib/src/screens/logging/logging_controller.dart

```diff
@@ -548,40 +578,49 @@ class LoggingController extends DisposableController
   @override
   void filterData(Filter<LogData> filter) {
     super.filterData(filter);
-    final queryFilter = filter.queryFilter;
-    if (queryFilter.isEmpty) {
-      filteredData
-        ..clear()
-        ..addAll(data);
-      return;
-    }
-    filteredData
-      ..clear()
-      ..addAll(
-        data.where((log) {
-          final kindArg = filter.queryFilter.filterArguments[kindFilterId];
-          if (kindArg != null &&
-              !kindArg.matchesValue(log.kind.toLowerCase())) {
-            return false;
-          }
 
-          if (filter.queryFilter.substrings.isNotEmpty) {
-            for (final substring in filter.queryFilter.substrings) {
-              final matchesKind = log.kind.caseInsensitiveContains(substring);
-              if (matchesKind) return true;
+    bool filterCallback(LogData log) {
+      for (final toggleFilter in filter.toggleFilters) {
```

Comment:
Could we set this check to a variable to make it easier to understand what it's checking for?

Something like (might be a better way to write this):
```
final matchesToggledFilter = filter.toggleFilters.any(toggleFilter.enabled.value && !toggleFilter.includeCallback(log));
if (matchesToggledFilter) return false;
```